plugin/trino-postgresql/src/test/java/io/trino/plugin/postgresql/TestingPostgreSqlServer.java

```diff
@@ -88,6 +93,18 @@ private static void execute(String url, Properties properties, String sql)
         }
     }
 
+    DatabaseEvents startCollecting()
+    {
+        return new DatabaseEvents(this);
+    }
+
+    DatabaseEvents recordEventsForOperations(Runnable operation)
+    {
+        DatabaseEvents events = startCollecting();
+        operation.run();
+        return events;
```

Comment:
This object is mutable. So if somebody would wait with calling `streamQueriesContaining` then they may see different (more) results. Maybe we should materialize it now and return materialized events?

Reply:
Ok I can make it immutable - it's for sure a better design.